From: Andrew Vagin <avagin@openvz.org>
Date: Mon Sep 26 19:55:32 2011 +0400
Subject: perf: Fix counter of ftrace events
References: bnc#789200, fate#313753, fate#312066, fate#313762
Git-commit: 92e51938f5d005026ba4bb5b1fae5a86dc195b86
Patch-mainline: v3.2-rc1
Signed-off-by:  Tony Jones <tonyj@suse.de>

    Each event adds some points to its counters. By default it adds 1,
    and a number of points may be transmited in event's parameters.
    
    E.g. sched:sched_stat_runtime adds how long process has been running.
    
    But this functionality was broken by v2.6.31-rc5-392-gf413cdb
    and now the event's parameters doesn't affect on a number of points.
    
    TP_perf_assign isn't defined, so __perf_count(c) isn't executed and
    __count is always equal to 1.
    
    Signed-off-by: Andrew Vagin <avagin@openvz.org>
    Signed-off-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Link: http://lkml.kernel.org/r/1317052535-1765247-2-git-send-email-avagin@openvz.org
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff --git a/include/trace/ftrace.h b/include/trace/ftrace.h
index 533c49f..7697249 100644
--- a/include/trace/ftrace.h
+++ b/include/trace/ftrace.h
@@ -711,6 +711,9 @@ __attribute__((section("_ftrace_events"))) *__event_##call = &event_##call
 #undef __perf_count
 #define __perf_count(c) __count = (c)
 
+#undef TP_perf_assign
+#define TP_perf_assign(args...) args
+
 #undef DECLARE_EVENT_CLASS
 #define DECLARE_EVENT_CLASS(call, proto, args, tstruct, assign, print)	\
 static notrace void							\
